fairscale/experimental/nn/sync_batchnorm.py

```diff
@@ -62,6 +40,87 @@ def _forward(
     _forward = torch.jit.script(_forward)  # type: ignore
 
 
+class _SyncBatchNormFunction(torch.autograd.Function):
+    @staticmethod
+    # type: ignore
+    def forward(
+        ctx, input, weight, bias, affine, track_running_stats, running_mean, running_var, eps, momentum, process_group
+    ):
+        dim = [d for d in range(input.ndim) if d != 1]
+        count = torch.full((1,), input.numel() // input.size(1), device=input.device, dtype=input.dtype)
+        total_count = count.clone()
+        handle = dist.all_reduce(total_count, group=process_group, async_op=True)
+        mean = torch.mean(input, dim=dim, keepdim=True)
+        meansqr = torch.mean(input * input, dim=dim, keepdim=True)
```

Comment:
it seems that input.pow(2) is faster?

Reply:
Cool. Do you have a reference?

Reply:
I don’t have one. I just did some test in ipython which seems faster but that could be noise.